### PR TITLE
Emulated 3D textures

### PR DESCRIPTION
**Pull request by izaid**

### CODE REVIEW

**Comment by Eric89GXL** [comment]
There are no tests for the new class, but there are tests for all of the others. Can you add similar ones?

File: vispy/gloo/texture.py [hunk 1]
```diff
@@ -392,6 +392,18 @@ def glsl_type(self):
...
 
+    @property
+    def glsl_sampler_type(self):
+        """ GLSL type of the sampler.
+        """
+        return 'sampler1D'
+
+    @property
+    def glsl_sample(self):
+        """ GLSL function that samples the the texture.
```
Comment (almarklein) [inline vispy/gloo/texture.py]: the the

File: vispy/gloo/texture.py [hunk 2]
```diff
@@ -433,6 +445,18 @@ def glsl_type(self):
...
 
+    @property
+    def glsl_sampler_type(self):
+        """ GLSL type of the sampler.
+        """
+        return 'sampler2D'
+
+    @property
+    def glsl_sample(self):
+        """ GLSL function that samples the the texture.
```
Comment (almarklein) [inline vispy/gloo/texture.py]: the the

File: vispy/gloo/texture.py [hunk 3]
```diff
@@ -476,6 +500,64 @@ def glsl_type(self):
...
 
+    @property
+    def glsl_sampler_type(self):
+        """ GLSL type of the sampler.
+        """
+        return 'sampler3D'
+
+    @property
+    def glsl_sample(self):
+        """ GLSL function that samples the the texture.
```
Comment (almarklein) [inline vispy/gloo/texture.py]: the the (think this was the last none :P)

**Comment by almarklein** [comment]
Apart from minor comments, LGTM. We probably want a mechanism to make the VolumeVisual select this class instead of Texture3D when es2 is used. You want to look at that in this PR, or do it in a new one?

File: vispy/gloo/texture.py [hunk 4]
```diff
@@ -476,6 +500,64 @@ def glsl_type(self):
...
+        data = np.transpose(data, (1, 0, 2, 3)) # This shouldn't be necessary, but we need to rewrite _glsl_sample
+        data = np.reshape(data, (data.shape[0], data.shape[1] * data.shape[2], 3))
+        Texture2D.__init__(self, data, format, **kwargs)
+
+        self._glsl_sample = Function(self.__class__._glsl_sample)
+        self._glsl_sample['depth'] = depth
+
+    @property
+    def glsl_sample(self):
+        """ GLSL function that samples the the texture.
```
Comment (QuLogic) [inline vispy/gloo/texture.py]: Sorry, @almarklein, I think you missed one. :wink:

**Pull request closed without merging**